apps/cli/src/commands/LaunchUpdate.ts

```diff
@@ -0,0 +1,193 @@
+
+  await downloadAndInstallLatestDevBuildAsync({
+    deviceId,
+    manifest,
+    platform: AppPlatform.Android,
+    distribution: DistributionType.Internal,
+  });
+  await Emulator.openURLAsync({ url: getUpdateDeeplink(updateURL, manifest), pid: device.pid });
+}
+
+async function launchUpdateOnIOSAsync(updateURL: string, manifest: Manifest, deviceId: string) {
+  const isSimulator = await Simulator.isSimulatorAsync(deviceId);
+  if (!isSimulator) {
+    throw new Error('Launching updates on iOS physical is not supported yet');
+  }
+
+  await downloadAndInstallLatestDevBuildAsync({
+    deviceId,
+    manifest,
+    platform: AppPlatform.Ios,
+    distribution: DistributionType.Simulator,
+  });
+
+  await Simulator.openURLAsync({
+    url: getUpdateDeeplink(updateURL, manifest),
+    udid: deviceId,
+  });
+}
+
+function getExpoGoUpdateDeeplink(updateURL: string, manifest: Manifest) {
+  if (updateURL.startsWith('https://u.expo.dev')) {
+    return `exp://u.expo.dev/update/${manifest.id}`;
+  }
+  return updateURL.replace('https://', 'exp://');
+}
+
+function getUpdateDeeplink(updateURL: string, manifest: Manifest) {
+  const updateIdURL = updateURL.startsWith('https://u.expo.dev')
+    ? `https://u.expo.dev/update/${manifest.id}`
+    : updateURL;
+
+  const scheme = Array.isArray(manifest?.extra?.expoClient?.scheme)
+    ? manifest?.extra?.expoClient?.scheme[0]
+    : manifest?.extra?.expoClient?.scheme;
+  scheme;
+  const slug = manifest?.extra?.expoClient?.slug;
+
+  if (!scheme && !slug) {
+    throw new Error('Unable to resolve schema from manifest');
+  }
+
+  return `${scheme || `exp+${slug}`}://expo-development-client/?url=${updateIdURL}`;
+}
+
+async function downloadAndInstallLatestDevBuildAsync({
+  deviceId,
+  manifest,
+  platform,
+  distribution,
+}: {
+  deviceId: string;
+  manifest: Manifest;
+  platform: AppPlatform;
+  distribution: DistributionType;
+}) {
+  const buildArtifactsURL = await getBuildArtifactsURLForUpdateAsync({
+    manifest,
+    platform,
+    distribution,
+  });
+  const buildLocalPath = await downloadBuildAsync(buildArtifactsURL);
+
+  if (platform === AppPlatform.Ios) {
+    await Simulator.installAppAsync(deviceId, buildLocalPath);
+  } else {
+    const device = await Emulator.getRunningDeviceAsync(deviceId);
+    await Emulator.installAppAsync(device, buildLocalPath);
+  }
+}
+
+async function getBuildArtifactsURLForUpdateAsync({
+  manifest,
+  platform,
+  distribution,
+}: {
+  manifest: Manifest;
+  platform: AppPlatform;
+  distribution: DistributionType;
+}): Promise<string> {
+  const { app } = await graphqlSdk.getAppBuildForUpdate({
+    // TODO(gabrieldonadel): Add runtimeVersion filter
+    appId: manifest.extra?.eas?.projectId ?? '',
+    platform,
+    distribution,
+  });
+
+  const build = app?.byId?.buildsPaginated?.edges?.[0]?.node;
+  if (
+    build.__typename === 'Build' &&
+    build.expirationDate &&
+    new Date(build.expirationDate) > new Date() &&
```

Comment:
```suggestion
    new Date(build.expirationDate) > Date.now() &&
```
Not a problem at all, I just think it's clearer but up to you

Reply:
sure, makes sense to me